sklearn/decomposition/_kernel_pca.py

```diff
@@ -56,10 +63,31 @@ class KernelPCA(TransformerMixin, BaseEstimator):
         Learn the inverse transform for non-precomputed kernels.
         (i.e. learn to find the pre-image of a point)
 
-    eigen_solver : string ['auto'|'dense'|'arpack'], default='auto'
-        Select eigensolver to use. If n_components is much less than
-        the number of training samples, arpack may be more efficient
-        than the dense eigensolver.
+    eigen_solver : string {'auto', 'dense', 'arpack', 'randomized'}
+        Select eigensolver to use (default='auto'). If `n_components` is much
+        less than the number of training samples, randomized (or arpack to a
+        smaller extend) may be more efficient than the dense eigensolver.
+        Randomized SVD is performed according to the method of Halko et al.
+
+        auto :
+            the solver is selected by a default policy based on `K.shape[0]`
```

Comment:
```suggestion
            the solver is selected by a default policy based on n_samples
```

Reply:
Yes this is much more readable I agree !